crates/boojum/src/gadgets/u256/mod.rs

```diff
@@ -394,19 +394,21 @@ impl<F: SmallField> UInt256<F> {
         let q = UInt256::allocate(cs, q);
         let r = UInt256::allocate(cs, r);
 
-        let (_, m_greater_than_r) = r.overflowing_sub(cs, &modulo);
+        let mod_is_zero = Boolean::allocate(cs, m.is_zero());
```

Comment:
you can't do this. Allocate just m and then check with constraint is_zero this 'm'